### DISCUSSION TITLE
Run tool/task directly after generation phase

### DISCUSSION THREAD

**Discussion by itavero** [microsoft/vscode-cmake-tools#4199] [Q&A] [answered]
Is there a way to run a tool/task after the generation phase of CMake using this extension (or CMake itself)?

I need to run a tool after the generation phase to extend the compilation database for some analysis tool.

**Comment by gcampbell-msft** [comment, marked as answer]
@itavero Yes, and no, depending on exactly what you want. We do provide a configure task, that is run through the tasks.json. Therefore, if you want to run your tool/task and have it depend on running a cmake configure, then it would first run configure and then run your task. 

However, I'm not sure there is a clean way to set up a task to always run after another task is ran.
